frontend/package.json

```diff
@@ -3,6 +3,13 @@
   "private": true,
   "repository": "https://github.com/Opetushallitus/kieli-ja-kaantajatutkinnot",
   "scripts": {
+    "g:tsc": "cd $INIT_CWD && tsc",
+    "g:eslint": "cd $INIT_CWD && eslint",
+    "g:stylelint": "cd $INIT_CWD && stylelint",
+    "g:prettier": "cd $INIT_CWD && prettier",
+    "g:jest": "cd $INIT_CWD && jest",
+    "g:webpack": "cd $INIT_CWD && webpack",
+    "g:cypress": "cd $INIT_CWD && cypress",
     "akt:start": "yarn workspace @oph-kieli-ja-kaantajatutkinnot/akt start",
```

Comment:
Tulisiko tähän samaan malliin muillekin projekteille suurin piirtein nämä samat skriptit, ts. esim `vkt:start`, `vkt:lint`, yms.

Reply:
Noita ei ole pakko laittaa siihen tällä tavalla. Voi ihan hyvin käyttää `yarn workspace ....` manuaalisesti, mutta mielestäni kuitenkin noilla skripteillä on helpompi tehdä devausta.

Reply:
Yksi vaihtoehto voisi olla että tuolla `frontend/packages/akt/package.json` alla skriptit alkaisivat `akt:`-prefiksillä. Silloin ei kai tarvitsisi enää täällä määritellä niitä uudestaan. Toki sitten ei voisi enää ajaa skriptejä ilman noita projektikohtaisia etuliitteitä.

Reply:
Mutta ylipäänsä en tiedä kuinka hyvä asia on määritellä samannimisiä skriptejä (ilman noita `akt/vkt/otr`-etuliitteitä) useassa workspacessa. Nehän näkyvät muihin workspaceihin (jos skriptin nimessä esiintyy `:`), jolloin voisi ehkä tulla sekaannuksia, jos näitä kaikkia tavanomaisia skriptejä ei olekaan määritelty kaikissa workspaceissa.

Reply:
Tosiaan noita frontend/packages alla näkyvät skriptit ovat lähinnä infra juttuja varten. Projektikohtaiset nimet projektin package.jsonissa ei sinänsä vaikuta tuohon, koska joutuu ajaamaan aina `workspace/projektinnimi  skriptin`.

Reply:
Lähinnä siis tarkoitan seuraavaa:
1.`frontend/package.json`: poista skripti `akt:format`
2.`frontend/packages/akt/package.json`: muuta skripti `format` -> `akt:format`

Näiden jälkeenhän `yarn akt:format` toimii kaikkialla `frontend/` alta ajettuna.

Reply:
Samoin nyt `frontend/` alta voidaan ajaa `yarn test:jest` ja se tällä hetkellä löytää `frontend/packages/akt/package.json` määritellyn skriptin, mutta en tiedä mitä tapahtuisi jos myös `frontend/packages/vkt/package.json` olisi määritelty samanniminen skripti. 

Mietin siis https://yarnpkg.com/getting-started/qa#how-to-share-scripts-between-workspaces pohjalta (ilman varsinaista laajempaa kokemusta yarnista) että voisi olla fiksua olla noi projektikohtaiset prefiksit kaikissa skripteissä.

Reply:
Toi on hyvä ehdotus. Jännää että se toimii :D Muistaakseni kokeilin sitä yarn 1.X ja silloin ei toiminut, mutta nyt ainakin 3.2:lla näyttäisi toimivan lokaalisti. Muutetaan tähän muotoon.

Reply:
Katsotaan mitä Github Actions sanoo siitä..

Reply:
Näyttäisi toimivan CI:ssäkin.